### DISCUSSION TITLE
Long schedule times with non-Cirrus windows containers.

### DISCUSSION THREAD

**Discussion by xddxd** [cirruslabs/cirrus-ci-docs#836] [Q&A]
Whenever I use my own windowsservercore container that has the latest visual studio build tools and python baked in, scheduling a task on Cirrus takes about 5-10 minutes, when I use Cirrus's windowsservercore:cmake container (which has visual studio 16.8, unfortunately not optimal for me), it gets scheduled within a minute. Why does this happen?

**Comment by fkorotkov** [comment]
Could you please provide an image that you use? 

The reason of long scheduling is that Windows images are usually very large and it takes time to download layers and unarchive them. We do cache some of the popular images (like the one you mentioned) which makes the process faster.

**Reply by xddxd** [reply]
Here you go.
[Link](https://hub.docker.com/layers/xddxd/windowsservercore/single/images/sha256-10da029b33127706c5b478fce42a14fdd846e9448c28754e6d9003a3fee595cc?context=explore)

**Reply by fkorotkov** [reply]
@xddxd I'm considering changing our image in https://github.com/cirruslabs/docker-images-windows/pull/22. Seems then you can base your image of ours and just install `python`. Will that work for you?

**Reply by xddxd** [reply]
Absolutely, thank you very much.
